### ISSUE TITLE
Error during msys2_update

### ISSUE THREAD

**Issue by jkburges** [brholtkamp/msys2-cookbook#6]
I get the following error when running default recipe, with default attributes:

```
msys2_update("update MSYS2") do
  action [:run]
  retries 0
  retry_delay 2
  default_guard_interpreter :default
  declared_type :msys2_update
  cookbook_name :msys2
  recipe_name "default"
  only_if { #code block }
end

<snip>

STDERR: error: failed to commit transaction (conflicting files)
 ---- End output of C:/msys64/usr/bin/bash.exe -l -c 'cd '/' && pacman --sync --sysupgrade --refresh --noconfirm --noprogressbar' ----
 Ran C:/msys64/usr/bin/bash.exe -l -c 'cd '/' && pacman --sync --sysupgrade --refresh --noconfirm --noprogressbar' returned 1
```

I *believe* the root cause is the same as mentioned in https://github.com/Alexpux/MSYS2-packages/issues/1024, but after reading that issue, I'm none the wiser as to what a "fix", or workaround, could be.

Any ideas?

**Comment by brholtkamp** [comment]
Did you already have the recipe deployed on the machine that's attempting to update?  Or is this occurring on the initial run?

**Comment by wbowling** [comment]
> Did you already have the recipe deployed on the machine that's attempting to update? Or is this occurring on the initial run?

It's running from scratch on a fresh machine each time and started failing around a week ago:
```
amazon-ebs: checking keyring...
amazon-ebs: checking package integrity...
amazon-ebs: loading package files...
amazon-ebs: checking for file conflicts...
amazon-ebs: coreutils: /usr/lib/coreutils/libstdbuf.so exists in filesystem
amazon-ebs: Errors occurred, no packages were upgraded.
amazon-ebs: STDERR: error: failed to commit transaction (conflicting files)
amazon-ebs: ---- End output of C:/msys64/usr/bin/bash.exe -l -c 'cd '/' && pacman --sync --sysupgrade --refresh --noconfirm --noprogressbar' ----
```

We disabled `auto_update` and ran the update with `--force` which completed successfully:
```ruby
msys2_execute "update MSYS2 with force" do
   command ["pacman ", "-Syyuu", "--noconfirm", "--noprogressbar", "--force"]
end
```

Not sure why it suddenly started failing as I'm pretty sure there were no changes on our end

**Comment by brholtkamp** [comment]
Yeah, I'm guessing it's a broken package on the MSYS2 end of things.  I've personally had to fix a package upgrade myself with MSYS2, so it unfortunately isn't terribly shocking.

How long have y'all been running this cookbook before the error cropped up?

**Comment by jkburges** [comment]
This is still happening FWIW. Changing https://github.com/brholtkamp/msys2-cookbook/blob/master/resources/update.rb#L11 to something like we're doing in our own recipe, e.g.

```ruby
  msys2_execute 'update MSYS2' do
    command ["pacman ", "-Syyuu", "--noconfirm", "--noprogressbar", "--force", "--ask 20"]
  end
```

would probably fix it, but I'm not sure what the implications of those different options are, especially the `--force` one.

**Comment by brholtkamp** [comment]
I honestly have no clue if `--ask 20` is redundant with `--noconfirm`, but I don't think we'd want to have `--force` as a default as per the description on the man pages: https://www.archlinux.org/pacman/pacman.8.html.  We could however, make it an option for the cookbook that y'all could add to the role so it won't require manual alteration and deployment of the cookbook on y'alls end.
